src/Compilers/CSharp/Portable/Errors/MessageID.cs

```diff
@@ -323,6 +323,21 @@ public static LocalizableErrorArgument Localize(this MessageID id)
             return true;
         }
 
+        internal static bool CheckFeatureAvailability(
+            this MessageID feature,
+            DiagnosticBag diagnostics,
+            SyntaxNode syntax,
+            Location? location = null)
+        {
+            var diag = GetFeatureAvailabilityDiagnosticInfo(feature, (CSharpParseOptions)syntax.SyntaxTree.Options);
+            if (diag is object)
+            {
+                diagnostics.Add(diag, location ?? syntax.GetLocation());
+                return false;
+            }
+            return true;
+        }
+
         internal static bool CheckFeatureAvailability(
```

Comment:
The comments in method `RequiredVersion` below (line 375) should be updated too, from `// syntax check` or blank to `// semantic check`.

Reply:
done.

Reply:
Maybe we could remove most of those comments once all the checks are migrated. Then we should leave a comment there to not add parser checks when possible.

Reply:
Sure.  I can do that.